packages/labs/ssr-dom-shim/src/element-internals.ts

```diff
@@ -0,0 +1,164 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/**
+ * TODO
+ * - This type could be better inferred as Record<keyof ARIAMixin, string>;
+ *   however, modern browsers and TypeScript seem to lack a common
+ *   definition of the keys listed in ARIAMixin
+ */
+export const ariaMixinEnum: Record<string, string> = {
+  ariaAtomic: 'aria-atomic',
+  ariaAutoComplete: 'aria-autocomplete',
+  ariaBraileLabel: 'aria-brailelabel',
+  ariaBraileDescription: 'aria-brailedescription',
+  ariaBusy: 'aria-busy',
+  ariaChecked: 'aria-checked',
+  ariaColCount: 'aria-colcount',
+  ariaColIndex: 'aria-colindex',
+  ariaColSpan: 'aria-colspan',
+  ariaCurrent: 'aria-current',
+  ariaDescription: 'aria-description',
+  ariaDisabled: 'aria-disabled',
+  ariaExpanded: 'aria-expanded',
+  ariaHasPopup: 'aria-haspopup',
+  ariaHidden: 'aria-hidden',
+  ariaInvalid: 'aria-invalid',
+  ariaKeyShortcuts: 'aria-keyshortcuts',
+  ariaLabel: 'aria-label',
+  ariaLevel: 'aria-level',
+  ariaLive: 'aria-live',
+  ariaModal: 'aria-modal',
+  ariaMultiLine: 'aria-multiline',
+  ariaMultiSelectable: 'aria-multiselectable',
+  ariaOrientation: 'aria-orientation',
+  ariaPlaceholder: 'aria-placeholder',
+  ariaPosInSet: 'aria-posinset',
+  ariaPressed: 'aria-pressed',
+  ariaReadOnly: 'aria-readonly',
+  ariaRequired: 'aria-required',
+  ariaRoleDescription: 'aria-roledescription',
+  ariaRowCount: 'aria-rowcount',
+  ariaRowIndex: 'aria-rowindex',
+  ariaRowSpan: 'aria-rowspan',
+  ariaSelected: 'aria-selected',
+  ariaSetSize: 'aria-setsize',
+  ariaSort: 'aria-sort',
+  ariaValueMax: 'aria-valuemax',
+  ariaValueMin: 'aria-valuemin',
+  ariaValueNow: 'aria-valuenow',
+  ariaValueText: 'aria-valuetext',
+  role: 'role',
+};
+
+/**
+ * Reflect internals AOM attributes back to the DOM prior to hydration
+ * to ensure search bots can accurately parse element semantics prior
+ * to hydration. This is called whenever an instance of ElementInternals
+ * is created on an element to wire up the getters/setters
+ * for the AriaMixin properties
+ *
+ * TODO - Determine the proper way to hydrate any attributes set by the shim
```

Comment:
@justinfagnani Following up on your question from the other day. I'm happy to reference an issue, but I'm not sure where to put that. The version of TypeScript I'm running in VSCode matches the spec, but neither Chromium or Webkit (or Firefox with the ARIA reflection tag) have the same tags. The ones listed here are in line with Chromium's implementation. 

To complicate that a bit more, the version of TypeScript building Lit's repo is different from the version running in VSCode (somehow) so that it recognizes different properties on `ARIAMixin` which appears to be somewhat volatile in TypeScript. 

Any guidance you could provide on how to add more clarity here would be welcomed.